File: 8. Afficher.go
package main

import (
	"fmt"
	"io/ioutil"
	"sort"
)

type FileSize struct {
	Name string
	Size int64
}

func main() {
	folder := `C:\Users\Public\TP_Automatisation\test_logs`
	files, _ := ioutil.ReadDir(folder)
	var fs []FileSize
	for _, f := range files {
		if !f.IsDir() {
			fs = append(fs, FileSize{f.Name(), f.Size()})
		}
	}
	sort.Slice(fs, func(i, j int) bool { return fs[i].Size > fs[j].Size })
	for i, f := range fs {
		if i >= 5 {
			break
		}
		fmt.Printf("%s : %d octets\n", f.Name, f.Size)
	}
}

File: 8. Afficher.py
import os

folder = r"C:\Users\Public\TP_Automatisation\test_logs"
files = [(f, os.path.getsize(os.path.join(folder, f))) for f in os.listdir(folder) if os.path.isfile(os.path.join(folder, f))]
files.sort(key=lambda x: x[1], reverse=True) 
for f, size in files[:5]:
    print(f"{f} : {size} octets")
